include/wamr/WAMRWasmModule.h

```diff
@@ -45,6 +39,14 @@ class WAMRWasmModule final : public WasmModule
     // ----- Module lifecycle -----
     void doBindToFunction(faabric::Message& msg, bool cache) override;
 
+    void writeStringArrayToMemory(const std::vector<std::string>& strings,
```

Comment:
Unfortunately, I could not re-use the prototypes in `WasmModule.h` as the parameter types did not match. IMO I'd delete those from there, and leave two different implementations in each `XYZWasmModule` but up for discussion.